### PR TITLE
Param type documentation consistency updates.

### PR DESCRIPTION
**Pull request by paulidale**
Use the parameter types when describing generic params.

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: doc/man3/EVP_DigestInit.pod [hunk 1]
```diff
@@ -339,12 +339,13 @@ EVP_MD_CTX_set_params() can be used with the following OSSL_PARAM keys:
 
 =over 4
 
-=item OSSL_PARAM_DIGEST_KEY_XOFLEN <size_t>
+=item OSSL_PARAM_DIGEST_KEY_XOFLEN <unsigned integer>
 
 Sets the digest length for extendable output functions.
-It is used by the SHAKE algorithm.
+It is used by the SHAKE algorithm and should not exceed what can be given
+using a size_t.
```
Comment (levitte) [inline doc/man3/EVP_DigestInit.pod]: [man-pages(7)](http://man7.org/linux/man-pages/man7/man-pages.7.html) says something about types being bold, so `B<size_t>`

File: doc/man3/EVP_KDF.pod [hunk 2]
```diff
@@ -189,15 +189,16 @@ Some KDF implementations require a key.
...
 
-=item B<OSSL_KDF_PARAM_MAC_SIZE> ("maclen") <size_t>
+=item B<OSSL_KDF_PARAM_MAC_SIZE> ("maclen") <unsigned integer>
 
 Used by implementations that use a MAC with a variable output size (KMAC).
 For those KDF implementations that support it, this parameter
 sets the MAC output size.
 
 The default value, if any, is implementation dependent.
+The length must never exceed what can be given with a size_t.
```
Comment (levitte) [inline doc/man3/EVP_KDF.pod]: Same here...

File: doc/man3/EVP_KDF.pod [hunk 3]
```diff
@@ -208,6 +209,7 @@ If this memory usage limit is exceeded because the load factors are chosen
 too high, the key derivation will fail.
 
 The default value is implementation dependent.
+The memory size must never exceed what can be given with a size_t.
```
Comment (levitte) [inline doc/man3/EVP_KDF.pod]: ... and here

File: doc/man3/EVP_MAC.pod [hunk 4]
```diff
@@ -234,7 +234,8 @@ or SHAKE256.
 
 For MAC implementations that support it, set the output size that
 EVP_MAC_final() should produce.
-The allowed sizes vary between MAC implementations.
+The allowed sizes vary between MAC implementations, but must never exceed
+what can be given with a size_t.
```
Comment (levitte) [inline doc/man3/EVP_MAC.pod]: ... and here

File: doc/man3/EVP_DigestInit.pod (hunk 1 above)
Comment (paulidale) [reply doc/man3/EVP_DigestInit.pod]: Okay, I wasn't sure what the guideline was.
All changed now.

**Comment by levitte** [comment]
> Okay, I wasn't sure what the guideline was.

Depending on where you read, that are incredibly vague, and often expressed in terms of "for types, use this mdoc macro" or something like that, and you have to figure out from the \*roff language what it does...  I've found that man-pages(7) and perlpodstyle are the clearest docs on the matter, so I often turn to them.

(I sometimes look at my old copy of "UNIX Programmer's Manual Volume 1" from 1982, but not too deeply, as it seems the conventions have changed since then)

File: doc/man3/EVP_KDF.pod [hunk 5]
```diff
@@ -208,6 +209,7 @@ If this memory usage limit is exceeded because the load factors are chosen
 too high, the key derivation will fail.
 
 The default value is implementation dependent.
+The memory size must never exceed what can be given with a B<size_t>.
```
Comment (richsalz) [inline doc/man3/EVP_KDF.pod]: Is this enforced by the code?

Comment (levitte) [reply doc/man3/EVP_KDF.pod]: It's enforced enough by the way it's used

**Comment by paulidale** [comment]
Merged, thanks.

**Pull request closed without merging**